layers/+lang/csharp/funcs.el

```diff
@@ -0,0 +1,150 @@
+    ;; [missing in roslyn] (spacemacs/declare-prefix-for-mode 'csharp-mode "mf" "csharp/file")
+    (spacemacs/declare-prefix-for-mode 'csharp-mode "mg" "csharp/navigation")
+    (spacemacs/declare-prefix-for-mode 'csharp-mode "mh" "csharp/documentation")
+    (spacemacs/declare-prefix-for-mode 'csharp-mode "mr" "csharp/refactoring")
+    (spacemacs/declare-prefix-for-mode 'csharp-mode "ms" "csharp/server")
+    (spacemacs/declare-prefix-for-mode 'csharp-mode "mt" "csharp/tests")
+
+    (spacemacs/set-leader-keys-for-major-mode 'csharp-mode
+      ;; Compile
+      ;; Only one compile command so use top-level
+      ;; [missing in roslyn] "cc" 'omnisharp-build-in-emacs
+
+      ;; Solution/project manipulation
+      ;; [missing in roslyn] "fa" 'omnisharp-add-to-solution-current-file
+      ;; [missing in roslyn] "fA" 'omnisharp-add-to-solution-dired-selected-files
+      ;; [missing in roslyn] "fr" 'omnisharp-remove-from-project-current-file
+      ;; [missing in roslyn] "fR" 'omnisharp-remove-from-project-dired-selected-files
+
+      ;; [missing in roslyn] "pl" 'omnisharp-add-reference
+
+      ;; Navigation
+      "ge"   'omnisharp-solution-errors
+      "gG"   'omnisharp-go-to-definition-other-window
+      "gu"   'omnisharp-helm-find-usages
+      "gU"   'omnisharp-find-usages-with-ido
+      "gs"   'omnisharp-helm-find-symbols
+      "gi"   'omnisharp-find-implementations
+      "gI"   'omnisharp-find-implementations-with-ido
+      "gr"   'omnisharp-navigate-to-region
+      "gm"   'omnisharp-navigate-to-solution-member
+      "gM"   'omnisharp-navigate-to-solution-member-other-window
+      "gf"   'omnisharp-navigate-to-solution-file
+      "gF"   'omnisharp-navigate-to-solution-file-then-file-member
+      "gc"   'omnisharp-navigate-to-current-file-member
+
+      ;; Help, documentation, info
+      "ht" 'omnisharp-current-type-information
+      "hT" 'omnisharp-current-type-information-to-kill-ring
+
+      ;; Refactoring
+      "rm" 'omnisharp-rename
+      ;; [Broken in roslyn] "rM" 'omnisharp-rename-interactively
+      "rr" 'omnisharp-run-code-action-refactoring
+
+      ;; Server manipulation, inspired spacemacs REPL bindings since C# does
+      ;; not provice a REPL
+      "ss" 'omnisharp-start-omnisharp-server
+      "sS" 'omnisharp-stop-server
+      "sr" 'omnisharp-reload-solution
+      "si" 'omnisharp-install-server
+
+      ;; Tests
+      ;; [missing in roslyn] "ta" 'omnisharp-unit-test-all
+      "tb" 'omnisharp-unit-test-buffer
+      "tl" 'omnisharp-unit-test-last
+      "tt" 'omnisharp-unit-test-at-point
+
+      ;; Code manipulation
+      "u" 'omnisharp-auto-complete-overrides
+      "i" 'omnisharp-fix-usings
+      ;; [missing in roslyn] "=" 'omnisharp-code-format
+      )
+    (spacemacs|hide-lighter omnisharp-mode)))
+
+
+;; lsp
+
+(defun spacemacs//csharp-setup-lsp ()
+  "Setup lsp backend."
+  (when (configuration-layer/layer-used-p 'lsp)
```

Comment:
Please remove all tests against the usage of lsp layer in your code. You request lsp to be loaded in your layer file so you can be sure that it is available at the point of loading. So all these tests will be true.

Reply:
OK